ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/UpdatePartColStatHandler.java

```diff
@@ -35,7 +38,32 @@ UpdatePartitionColumnStatMessage eventMessage(String stringRepresentation) {
 
   @Override
   public void handle(Context withinContext) throws Exception {
-    LOG.info("Processing#{} UpdateTableColumnStat message : {}", fromEventId(), eventMessageAsJSON);
+    LOG.info("Processing#{} UpdatePartitionTableColumnStat message : {}", fromEventId(),
+            eventMessageAsJSON);
+
+    org.apache.hadoop.hive.metastore.api.Table tableObj = eventMessage.getTableObject();
+    if (tableObj == null) {
+      LOG.debug("Event#{} was an event of type {} with no table listed", fromEventId(),
+              event.getEventType());
+      return;
+    }
+
+    if (!Utils.shouldReplicate(withinContext.replicationSpec, new Table(tableObj),
+                              withinContext.hiveConf)) {
+      return;
+    }
+
+    // Statistics without any data does not make sense.
+    if (withinContext.replicationSpec.isMetadataOnly()) {
```

Comment:
Metadata-only can be checked earlier to avoid heavier operations like Utils.shouldReplicate().

Reply:
Yes agreed. Done. I have also moved the transactional table check earlier since that's also cheaper than shouldReplicate().